datafusion/sql/src/expr/value.rs

```diff
@@ -405,11 +405,29 @@ const fn try_decode_hex_char(c: u8) -> Option<u8> {
 }
 
 /// Parse Decimal128 from a string
-///
-/// TODO: support parsing from scientific notation
 fn parse_decimal_128(unsigned_number: &str, negative: bool) -> Result<Expr> {
     // remove leading zeroes
     let trimmed = unsigned_number.trim_start_matches('0');
+
+    // check if the number is scientific notation
+    let parts = trimmed.split(|c| c == 'e' || c == 'E').collect::<Vec<_>>();
```

Comment:
It seems like `BigDecimal::from_str` handles scientific notation -- https://docs.rs/bigdecimal/latest/bigdecimal/struct.BigDecimal.html#examples

Since it is already a dependency, perhaps we can switch to using its implementation of parsing?

Reply:
Good idea